src/riak_kv_qry_compiler.erl

```diff
@@ -641,6 +663,7 @@ break_out_timeseries(Filters1, LocalFields1, [QuantumFields]) ->
         {_, {{_,_,{_,Starts}}, {_,_,{_,Ends}}}} when is_integer(Starts),
                                                      is_integer(Ends),
                                                      Starts > Ends ->
+            %% FIXME reliance on three element key!?
```

Comment:
Fixme still here - and it does look like it relies on the three element key structure.